ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.KsqlEngine;
+import io.confluent.ksql.metastore.KsqlTopic;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import io.confluent.ksql.util.ExecutorUtil;
+import io.confluent.ksql.util.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ClusterTerminator {
+
+  private final KsqlConfig ksqlConfig;
+  private final KsqlEngine ksqlEngine;
+
+  public ClusterTerminator(
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine
+  ) {
+    Objects.requireNonNull(ksqlConfig, "ksqlConfig is null.");
+    Objects.requireNonNull(ksqlEngine, "ksqlEngine is null.");
+    this.ksqlConfig = ksqlConfig;
+    this.ksqlEngine = ksqlEngine;
+  }
+
+  @SuppressWarnings("unchecked")
+  public void terminateCluster(final List<String> deleteTopicPatterns) {
+    validateDeleteList(deleteTopicPatterns);
+    terminatePersistentQueries();
+    deleteSinkTopics(deleteTopicPatterns);
+    deleteCommandTopic();
+    ksqlEngine.close();
+  }
+
+  private void terminatePersistentQueries() {
+    new ArrayList<>(ksqlEngine.getPersistentQueries()).stream()
+        .map(PersistentQueryMetadata::getQueryId)
+        .forEach(queryId -> ksqlEngine.terminateQuery(queryId, true));
+  }
+
+  private void validateDeleteList(final List<String> deleteTopicPatterns) {
```

Comment:
But you currently fail if the user supplies a regex that matches any non-sink topics, e.g. if I want all sink topics deleted then I'll supply the regex `.*`. But this will fail because it matches non-sink topics too.  This is bad.

I guess what we _do_ want to fail is a literal pattern, i.e. one without any special regex characters in it, that matches a non-sink topic.  To do this you'll need a regex to determine if each pattern is a literal or a regex.

However, I don't think its worth the complexity.  If we ensure the documentation is clear that the pattern is used to match against _sink topics only_, then we can remove this check.

Ensure you've added test case(s) to ensure non-sink topics that are matched as _not_ deleted.